sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -150,6 +154,49 @@
  *     ...
  * }</pre>
  *
+ * <p>Kafka provides deserializers for common types in
+ * {@link org.apache.kafka.common.serialization}. Alternatively, to use Kafka deserializers
+ * based on wrapping Beam {@link Coder}s, use {@code readWithCoders} instead of {@code read}:
```

Comment:
Add a note here that this is discouraged because the wire format of a Coder is not guaranteed, so this will only work if the data is exchanged with another Beam pipeline using KafkaIO with the same coder?